### PR TITLE
make more strict matching between events and aggregated objects

### PR DESCRIPTION
**Pull request by evasiba-ericsson**
### Applicable Issues
Previously if the rule for matching event to aggregated object failed to give the aggregated object but it returned id's of an event that have been already aggregated then the match would go through this event and therefore the event will aggregate to all this objects. This resulted in uncontrolled undesired merging to existing aggregations.

### Description of the Change
While matching incoming event to aggregated object remove the search in the map keeping track of the events. Matching will be solely based on the matchID rule.

### Alternate Designs
None.

### Benefits
Avoid unexpected information additions or overwritting in aggregated objects.

### Possible Drawbacks
None

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Vasile Baluta, vasile.baluta@ericsson.com

### CODE REVIEW

File: src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java [hunk 1]
```diff
@@ -294,10 +286,19 @@ private void setupRestEndpoints() {
...
+        while (!connected) {
+            try {
+                int port = SocketUtils.findAvailableTcpPort();
+                LOGGER.debug("Setting SMTP port to " + port);
+                mailSender.setPort(port);
+                smtpServer = SimpleSmtpServer.start(port);
+                // connected, go on
+                connected = true;
+            } catch (BindException e) {
+                // address already in use. Try again!
```
Comment (e-edling-ericsson) [inline src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: Maybe write something here if it goes wrong?

File: src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java [hunk 2]
```diff
@@ -154,23 +155,20 @@ private void updateJsonObject(JSONObject aggregatedJsonObject, JSONObject prepar
...
-                        // updateJsonObject(o1, o2);
-                    } else if (valueClass.equals(JSONArray.class)) {
-                        updateJsonObject((JSONArray) aggregatedJsonObject.get(preparedJsonKey),
-                                (JSONArray) preparedJsonObject.get(preparedJsonKey));
-                        // final JSONArray a1 = getAsT(aggregatedJsonObject,
-                        // preparedJsonKey, JSONArray.class);
-                        // final JSONArray a2 = getAsT(preparedJsonObject,
-                        // preparedJsonKey, JSONArray.class);
-                        // updateJsonObject(a1, a2);
+                    final Object eFromAgg = aggregatedJsonObject.get(preparedJsonKey);
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java]: What does this variable contain? I don't think eFromAgg is very descriptive.

File: src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java [hunk 3]
```diff
@@ -154,23 +155,20 @@ private void updateJsonObject(JSONObject aggregatedJsonObject, JSONObject prepar
...
+                    final Object eFromPrep = preparedJsonObject.get(preparedJsonKey);
```
Comment (e-edling-ericsson) [inline src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java]: Same with this

File: src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java [hunk 4]
```diff
@@ -115,25 +116,23 @@ public void in_MongoDb_RepeatFlagHandler_and_subscription_collections_the_subscr
...
         HttpRequest deleteRequest = new HttpRequest(HttpRequest.HttpMethod.DELETE);
-        ResponseEntity response = deleteRequest.setHost(getHostName())
-                .setPort(applicationPort)
-                .addHeader("content-type", "application/json")
-                .addHeader("Accept", "application/json")
-                .setEndpoint(subscriptionEndPoint + name)
-                .performRequest();
+        ResponseEntity response = deleteRequest.setHost(getHostName()).setPort(applicationPort)
+                .addHeader("content-type", "application/json").addHeader("Accept", "application/json")
+                .setEndpoint(subscriptionEndPoint + name).performRequest();
```
[the remark this answers is not in the document]
Comment (evasiba-ericsson) [reply src/functionaltests/java/com/ericsson/ei/subscriptions/repeatHandler/SubscriptionRepeatHandlerSteps.java]: Done

File: src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java (hunk 1 above)
Comment (evasiba-ericsson) [reply src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java]: Done

File: src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java (hunk 2 above)
Comment (evasiba-ericsson) [reply src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java]: Done

File: src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java (hunk 3 above)
Comment (evasiba-ericsson) [reply src/main/java/com/ericsson/ei/jsonmerge/MergeHandler.java]: Done

**Pull request merged**